src/lib.rs

```diff
@@ -509,15 +542,78 @@ impl FailPoint {
 /// Registry with failpoints configuration.
 type Registry = HashMap<String, Arc<FailPoint>>;
 
-#[derive(Debug, Default)]
-struct FailPointRegistry {
+/// Fail point registry. Threads bound to the same registry share the same
+/// failpoints configuration.
+#[derive(Debug, Default, Clone)]
+pub struct FailPointRegistry {
     // TODO: remove rwlock or store *mut FailPoint
-    registry: RwLock<Registry>,
+    registry: Option<Arc<RwLock<Registry>>>,
```

Comment:
I suggest to define a separate `FailPointRegistry` instead of using `Option` as field. In such way, code is cleaner and more readable.